checker/jdk/nullness/src/java/util/Objects.java

```diff
@@ -177,6 +177,7 @@
      * @return {@code obj} if not {@code null}
      * @throws NullPointerException if {@code obj} is {@code null}
      */
+    @EnsuresNonNull("#1")
     @SideEffectFree public static <T> T requireNonNull(@Nullable T obj) { throw new RuntimeException(); }
```

Comment:
It might as well. Done (again, in changes to be published shortly).

That said: Most of the time, the Checker Framework can probably infer `<T>` to be the appropriate non-null type, in which case the return value of plain `T` would be sufficient. Arguably we might even want to change the type parameter declaration to `<T extends @NonNull Object>` to ensure that this happens. But that might break any existing code that writes `Objects.<X>requireNonNull(x)`, where `<X>` has a nullable bound. (The code would have to change to `Objects.<@NonNull X>requireNonNull(x)`.) That seems fairly unlikely to come up, but maybe it's most consistent to accommodate such code anyway, similar to what you do with the type parameter bound for `Class` and `Optional`.

Reply:
> we don't overly mind keeping around stub files for cases in which we're willing to play a little more fast and loose with type-safety than the Checker Framework wants to by default.

I understand.  As much as possible, I would like to reduce your use of those stub files (or migrate some into the Checker Framework proper, as I proposed for the nullFriendlyCollections.astub).  One reason is to reduce your maintenance burden.  Another reason is to give other users the benefit of the annotation work you have done.  The conversations about the annotation tradeoffs are very useful, and I appreciate them.

Reply:
Hmm, I'm not sure how that comment ended up in this thread.  What I meant to say is that I'm inclined to keep the lenient bounds.  I'm often surprised by the imaginative ways that programmers use Java. :-)

Reply:
Coincidentally, I just came across a ton of calls to `Preconditions.<@Nullable Foo>checkNotNull(foo)` in our codebase. I assume they were a holdover from before we had a Checker Framework stub for the class, but still, that's evidence that tightening the bounds would force some users to make changes.